core/src/main/java/com/graphhopper/routing/HeadingResolver.java

```diff
@@ -26,10 +26,17 @@
 
 public class HeadingResolver {
     private final EdgeExplorer edgeExplorer;
-    private double toleranceRad = Math.toRadians(100);
+    private double toleranceRad;
 
     public HeadingResolver(Graph graph) {
         this.edgeExplorer = graph.createEdgeExplorer();
+        toleranceRad = Math.toRadians(100); //default tolerance
+    }
+
+    //Added constructor to allow configuration of the tolerance (issue #1646)
+    public HeadingResolver(Graph graph, int toleranceDec) {
+        this.edgeExplorer = graph.createEdgeExplorer();
+        toleranceRad = Math.toRadians(toleranceDec);
```

Comment:
Such comments, like "Added constructor..." are nice for a reviewer to understand what your attention was, but you can just add them here in github and they typically should not remain in the code directly. I mean stating your intention in comments can be a good thing, but for something simple like a new constructor it is not necessary, it is already clear when looking at the constructor parameters that this constructor allows setting the tolerance.

Another thing: It is usually a good idea to chain the constructors. Here the first one could just call the other like `this(graph, 100)` for example.